aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -972,19 +1017,58 @@ public void connect(
             final String customAuthorizer,
             final AWSIotMqttClientStatusCallback statusCallback) {
 
-        if (tokenKeyName == null || token == null || tokenSignature == null) {
+        if (tokenKeyName == null || token == null) {
             throw new IllegalArgumentException("tokenKeyName/token/tokenSignature cannot be null");
         }
 
         this.userStatusCallback = statusCallback;
 
+        final MqttConnectOptions options = new MqttConnectOptions();
+        // Set custom websocket header
+        customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Name", customAuthorizer);
+        customWebsocketHeaders.setProperty(tokenKeyName, token);
+        if (tokenSignature != null) {
+            customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Signature", tokenSignature);
+        }
+        customAuthConnect(options);
```

Comment:
This is a little confusing. You make an `options`, do some other stuff that doesn't touch `options`, and then call `customAuthConnect(options)`. 

Does that method use the headers you've just setup? To make that fact more clear, I wonder if you can just pass the map into the `customAuthConnect(...)` method, as a param?

Or better, should the headers be affixed into the `options`, somehow? Like maybe you do `options.setCustomWebsockHeaders(Map<String, String> headers)`, and then proceed to pass the `options` as-is, to the `customAuthConnect(` method?